test_strategy.md

```diff
@@ -0,0 +1,57 @@
+# Test strategy
+
+## Design test
+These are the checks and tests expected to be provided and maintained by the
+developers of a given operator. This is intentionally focusing on the
+bottom of the
+[test pyramid](https://martinfowler.com/articles/practical-test-pyramid.html).
+
+1. A common set of static analysis tools, stylers, linters aggregated into a
+single [pre-commit](https://pre-commit.com/) run both locally and in CI:
+   * go standard tools: fmt, vet, mod tidy
+   * [golangci-lint](https://golangci-lint.run/usage/linters/): a meta linter
+   with a tons of linters included but only a small subset enabled by default
+   * [operator-lint](https://github.com/gibizer/operator-lint): a home grew
+   linter automating operator development and testing specific rules
+
+   Each operator can extend the list of linters but we need a common minimum
+   set across each operators.
+
+2. Unit tests covering functions or types using go’s standard
+   [testing](https://go.dev/doc/tutorial/add-a-test) lib and
+   [gomega](https://onsi.github.io/gomega/)
+
+3. Component tests (also called functional or service tests) covering the
+   reconciliation of a single CRD or a set of CRDs from the same operator
+   using [EnvTest](https://book.kubebuilder.io/reference/envtest.html)
```

Comment:
I'm -2 on having the above tests expected at this point. I think it is fine for some groups to use these tests, but I don't yet want to be called out on having to write these. Of course there is some value in any tests... but it seems to me the investment in the EnvTest's in particular is quite high.

Reply:
The goal of this PR is not to guild anybody. But to agree on a common set of rules that, I think, would make both the product and the developer experience better.

Reply:
Thanks @gibizer for proposing it. Personally i found these tests useful as they provide faster responses and agree with the points already mentioned by You and Kamil.
I added these to some network operators already but not everyone in our DFG are familiar with it yet, same seems be true for other Teams too. As others have already planned work items for coming quarters expecting everyone to add these may not be right but can suggest and share the benefits of these so they can come to it whenever they think it's right time for them(atleast creating a Jira backlog would help teams so it's not forgotten fully).